tests/Unit/OAuthAuthenticatorTest.php

```diff
@@ -30,25 +35,91 @@ public function test_successful_authentication()
                 'resource_owner_id' => '1',
               ])
           );
-        $authenticator = new OAuthAuthenticator($mock);
-        $authenticationToken = $authenticator->authenticate('1234', '4567');
-        $this->assertEquals('MTQ0NjJkZmQ5OTM2NDE1ZTZjNGZmZjI3', $authenticationToken->getAccessToken());
+
+        /** @var SettingsRepository|MockInterface */
+        $settingsRepository = Mockery::mock(SettingsRepository::class);
+        $settingsRepository->allows(['getClientId' => 'client-id', "getClientSecret" => "client-secret"]);
+
+        $authenticator = new OAuthAuthenticator($authProvider, $settingsRepository);
+
+        $authenticationToken = $authenticator->authenticate();
+
+        $this->assertEquals('MTQ0NjJkZmQ5OTM2NDE1ZTZjNGZmZjI3', $authenticationToken->getToken());
     }
 
-    /**
-     * Test if the client can handle authentication failures
-     */
     public function test_failed_authentication()
     {
-        $mock = Mockery::mock(GenericProvider::class);
-        $mock->shouldReceive('getAccessToken')
-          ->andThrow(new IdentityProviderException('Invalid parameters.', 500, ''));
-        $authenticator = new OAuthAuthenticator($mock);
-        try {
-            $authenticator->authenticate('1234', '4567');
-            $this->assertEquals(true, false);
-        } catch (\Exception $e) {
-            $this->assertStringContainsString('Authentication failed', $e->getMessage());
-        }
+        $this->expectException(AuthenticationFailedException::class);
+
+        /** @var GenericProvider|MockInterface */
+        $authProvider = Mockery::mock(GenericProvider::class);
+        $authProvider->shouldReceive('getAccessToken')->andThrow(new IdentityProviderException('Invalid parameters.', 500, ''));
+
+        /** @var SettingsRepository|MockInterface */
+        $settingsRepository = Mockery::mock(SettingsRepository::class);
+        $settingsRepository->allows(['getClientId' => 'client-id', "getClientSecret" => "client-secret"]);
+
+        $authenticator = new OAuthAuthenticator($authProvider, $settingsRepository);
+
+        $authenticator->authenticate();
```

Comment:
Isn't there some kind of assertion missing here?

Reply:
Expect AuthenticationFailedException maybe?

Reply:
Exactly that assertion is at the beginning of the test (line 52).